File: receipt-analyzer-service/src/main/java/pl/madzierski/daniel/app/receipt/model/CreateReceiptRequest.java
package pl.madzierski.daniel.app.receipt.model;

import jakarta.validation.constraints.NotNull;
import pl.madzierski.daniel.app.receipt.scan_resolver.ReceiptResolverStrategyType;

import java.time.LocalDateTime;

public record CreateReceiptRequest(
        String name,
        String description,
        OCRHandlingResolver brand,
        LocalDateTime date,
        ReceiptResolverStrategyType strategy,
        @NotNull
        String walletId
) {
}

File: receipt-analyzer-service/src/main/kotlin/pl/madzierski/daniel/app/receipt/model/CreateReceiptRequest.kt
package pl.madzierski.daniel.app.receipt.model

import jakarta.validation.constraints.NotEmpty
import pl.madzierski.daniel.app.receipt.scan_resolver.ReceiptResolverStrategyType
import java.time.LocalDateTime

data class CreateReceiptRequest(
    @NotEmpty
    val name: String,
    val description: String?,
    val brand: OCRHandlingResolver?,
    val date: LocalDateTime?,
    val strategy: ReceiptResolverStrategyType,
)
